packages/flutter/lib/src/material/button_style_button.dart

```diff
@@ -16,6 +16,21 @@ import 'material.dart';
 import 'material_state.dart';
 import 'theme_data.dart';
 
+/// {@template flutter.material.ButtonStyleButton.iconAlignment}
+/// Determines alignment of icon i.e. after/before the label in TextButton.icon,
+/// ElevatedButton.icon, OutlinedButton.icon, FilledButton.icon,
+/// FilledButton.tonalIcon.
+///
+/// Defaults to [IconAlignment.left]
+/// {@endtemplate}
+enum IconAlignment {
+  /// icon is placed on the left side of the label
+  left,
```

Comment:
`left` -> `start` 
`right` -> `end` 

For compatibility with LTR and RTL, see https://main-api.flutter.dev/flutter/dart-ui/TextDirection.html

Reply:
I've a doubt, so we need to check TextDirection from theme you mean to say?

Reply:
Check for the current text direction (`final TextDirection textDirection = Directionality.of(context);`).

If the text direction is LTR and `IconAlignment.start` is set, it aligns the icon to the left and if RTL, to the right.
if the text direction is LTR and `IconAlignment.end` is set, it aligns to the to he right, and if RTL, to the left.

Reply:
Thanks a lot for this explanation, I couldn't understand at first. Appreciated 👍 🙌

Reply:
Implemented

Reply:
I was manually checking the textDirection too in code. I misunderstood your points. Reimplemented this suggestion.